doc/bucket-types.md

```diff
@@ -0,0 +1,41 @@
+Another feature of namespace buckets is cache, which can be configured to keep frequently-accessed objects in the system's memory, effectively reducing access time as well as egress and access costs.
+
+_1. In some cases, object metadata (such as tags) might have to be modified in order to comply with a cloud provider's limits_
+
+## Supported Storage Providers by Bucket Type
+| Service                                     | Data Buckets | Namespace Buckets |
+|---------------------------------------------|     :---:    |        :---:      |
+| Amazon Web Services S3 (incl. STS support)  | ✅          | ✅
+| S3 Compatible Services                      | ✅          | ✅
+| Azure Blob Storage                          | ✅          | ✅
+| Google Cloud Platform Object Storage        | ✅          | ❌
+| Filesystem (NSFS)                           | ❌          | ✅
+| Persistent Volume Pool                      | ✅          | ❌
+
+For specific API calls, you can check [AWS API Compatibility](https://github.com/noobaa/noobaa-core/blob/master/docs/design/AWS_API_Compatibility.md)
+
+_Please note that the table above might be out of date, or have additional support that is not present in older versions - the best way to check if a storage provider is supported is to run the NooBaa CLI tool (that fits the NooBaa version installed on the cluster) with a command such as `noobaa namespacestore create` or `noobaa backingstore create` and see which providers are listed in the help message._
+
+## Buckets in Relation to Bucketclasses and Stores
+In order to connect NooBaa to a storage provider (regardless of whether it's a cloud provider or an on-premises storage system), a store must be created - either a [backingstore](backing-store-crd.md), or a [namespacestore](namespace-store-crd.md). A store signifies a connection to a storage provider - it requires credentials, and sometimes additional configuartion such as a custom endpoint (when utilizing S3 compatible services that use the S3 API but aren't AWS).
+The type of store chosen determines what type of buckets will be created on top of it -
+- Namespacestores are used to create namespace buckets
+- Backingstores are used to create data buckets
+
+Once a store is created, it's necessary to create a [bucketclass](bucket-class-crd.md) resource in order to utilize it. Bucketclasses allow users to define bucket policies relating to data placement and replication, and are used as a middle layer between buckets and stores.
+
+After the bucketclass has been created, it's now finally possible to create an [object bucket claim](obc-provisioner.md), which is then reconciled by NooBaa and creates a bucket in the system. The created bucket can then be interacted with via the S3 API. For further reading on S3 API compatibility, see the [S3 Compatibility](s3-compatibility.md) document.
```

Comment:
1. Suggesting to add a new line.
2. If we have links to bucketclass and OBC, we can refer to this part.

```suggestion
After the bucketclass has been created, it's now finally possible to create an [object bucket claim](obc-provisioner.md), which is then reconciled by NooBaa and creates a bucket in the system. The created bucket can then be interacted with via the S3 API.

For further reading on S3 API compatibility, see the [S3 Compatibility](s3-compatibility.md) document.
```